src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -2637,11 +2669,42 @@ public IEnumerable<Document> GetResolvedDocumentsSince(DocumentsOperationContext
             }
         }
 
+        public string GetLastRevisionChangeVector(DocumentsOperationContext context)
+        {
+            var table = new Table(RevisionsSchema, context.Transaction.InnerTransaction);
+            var tvr = table.ReadLast(RevisionsSchema.FixedSizeIndexes[AllRevisionsEtagsSlice]);
+            if (tvr == null)
+                return null;
+            
+            return TableValueToChangeVector(context, (int)RevisionsTable.ChangeVector, ref tvr.Reader);
+        }
+
+        public string GetLastRevisionChangeVectorForCollection(DocumentsOperationContext context, string collection)
+        {
+            var table = GetOrCreateTable(context.Transaction.InnerTransaction, new CollectionName(collection));
+            var tvr = table.ReadLast(RevisionsSchema.FixedSizeIndexes[CollectionRevisionsEtagsSlice]);
+            return TableValueToChangeVector(context, (int)RevisionsTable.ChangeVector, ref tvr.Reader);
+        }
+
         public long GetNumberOfRevisionDocuments(DocumentsOperationContext context)
         {
             var table = new Table(RevisionsSchema, context.Transaction.InnerTransaction);
             return table.GetNumberOfEntriesFor(RevisionsSchema.FixedSizeIndexes[AllRevisionsEtagsSlice]);
         }
 
+        public long GetNumberOfRevisionDocumentsForCollection(DocumentsOperationContext context, string collection)
+        {
+            var table = GetOrCreateTable(context.Transaction.InnerTransaction, new CollectionName(collection));
+            return table.GetNumberOfEntriesFor(RevisionsSchema.FixedSizeIndexes[CollectionRevisionsEtagsSlice]);
+        }
+
+        internal Table GetOrCreateTable(Transaction tx, CollectionName collection)
```

Comment:
why not to use `EnsureRevisionTableCreated` instead ?

Reply:
@shaharhikri

Reply:
We talked about that, it doesn't matter on read

Reply:
EnsureRevisionTableCreated create the table if the cache doesn't contain it - which I don't want to do..